File: src/videoCourse_04/jav/lesson09/Loop.java
package videoCourse_04.jav.lesson09;

public class Loop {
    public static void main(String[] args) {
        for (int i = 0; i <= 100; i++) {
            System.out.println(i);
        }

        int[] array = new int[101];
        for (int i = 0; i < array.length; i++) {
            array[i] = i;
        }
        for (int i : array) {
            System.out.println(i);
        }
    }
}

File: src/videoCourse_04/kot/homework/lesson09/Loop.kt
package videoCourse_04.kot.homework.lesson09

fun main() {
    val array = arrayOfNulls<Int?>(301)

    for ((index, i) in (300..600).withIndex()) {
        array[index] = i
    }

    // for (i in (array.size - 1) downTo 0 step 5) { println(array[i]) } - тот же результат
    for (i in (array.size - 1) downTo 0) {
        if (array[i]?.rem(5) == 0) {
            println(array[i])
        }
    }
}